src/kyc.ts

```diff
@@ -56,10 +91,40 @@ export type PersonalDataAndDocumentsKyc = z.infer<
   typeof personalDataAndDocumentsKycSchema
 >
 
+export const personalDataAndDocumentsDetailedKycSchema =
+  personalDataAndDocumentsKycSchema
+    .omit({ identificationDocument: true })
+    .and(
+      z.object({
+        email: z.string().regex(EMAIL_REGEX),
+        identificationDocumentFront: z.string(),
+        identificationDocumentType: identificationDocumentTypeSchema,
+      }),
+    )
+    .and(
+      z
+        .object({
+          identificationDocumentType: identificationDocumentTypeWithBackSchema,
+          identificationDocumentBack: z.string(),
+        })
+        .or(
```

Comment:
If this `or` is "selected" by the parser (by `identificationDocumentBack` being omitted from the object), does the type of `identificationDocumentType` described here (`identificationDocumentTypeWithoutBackSchema`) override the more general one set above?

Reply:
That is, if `identificationDocumentBack` is missing, does this setup cause Zod to enforce that `identificationDocumentType` be of type `identificationDocumentTypeWithoutBackSchema`?

Reply:
to avoid having to reason about this I just dropped it. seems simpler.